rosidl_typesupport_opensplice_c/resource/srv__type_support_c.cpp.em

```diff
@@ -50,28 +63,28 @@ extern "C"
 
 // forward declare type support functions
 const rosidl_message_type_support_t *
-  ROSIDL_TYPESUPPORT_INTERFACE__MESSAGE_SYMBOL_NAME(rosidl_typesupport_opensplice_c, @(spec.pkg_name), @(subfolder), @(spec.srv_name)_Request)();
+  ROSIDL_TYPESUPPORT_INTERFACE__MESSAGE_SYMBOL_NAME(rosidl_typesupport_opensplice_c, @(', '.join(service.structure_type.namespaces)), @(service.structure_type.name)_Request)();
 const rosidl_message_type_support_t *
-  ROSIDL_TYPESUPPORT_INTERFACE__MESSAGE_SYMBOL_NAME(rosidl_typesupport_opensplice_c, @(spec.pkg_name), @(subfolder), @(spec.srv_name)_Response)();
+  ROSIDL_TYPESUPPORT_INTERFACE__MESSAGE_SYMBOL_NAME(rosidl_typesupport_opensplice_c, @(', '.join(service.structure_type.namespaces)), @(service.structure_type.name)_Response)();
 
 const char *
-register_types__@(subfolder)__@(spec.srv_name)(
+register_types__@('__'.join(service.structure_type.namespaces))__@(service.structure_type.name)(
```

Comment:
It looks like the symbol didn't include the package name before. Nothing against it, just wondering is there a technical reason requiring the package name here?

Reply:
There is no technical reason. These symbols should never leave the compilation unit. I think it was mostly for consistency with other implementations if I recall that correctly.